cloudflare.go

```diff
@@ -273,13 +286,14 @@ func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, u
 			}
 			break
 		}
-	}
+	} // for loop
```

Comment:
I added this comment to help understand where in the control flow the reader block was taking place. I actually think the body reading block should move out on control level -- that is, a successful API response breaks the for loop, then read the body.